Gordon360/Static Classes/Names.cs

```diff
@@ -121,8 +121,17 @@ public static class SQLQuery
         //ALL_ALUMNI_REQUEST returns the set of all the alumni in the records
         //Other methods query this set to find a specific alum based on AD_Username (or the encrypted id)
         public static string ALL_ALUMNI_REQUEST = "SELECT ID, WebUpdate, Title, FirstName, MiddleName, LastName, Suffix, MaidenName, NickName, HomeStreet1, HomeStreet2, HomeCity, HomeState, HomePostalCode, HomeCountry, HomePhone, HomeFax, HomeEmail, JobTitle, MaritalStatus, SpouseName, College, ClassYear, PreferredClassYear, Major1, Major2, ShareName, ShareAddress, Gender, GradDate, Email, grad_student, Barcode, ISNULL(AD_Username, CONVERT(NVARCHAR(32),HashBytes('SHA2_256', ID),2)) as AD_Username, show_pic, preferred_photo, Country, Major2Description, Major1Description from Alumni";// WHERE AD_Username is not null";
-        public static string ALL_BASIC_INFO_NOT_ALUM = "ALL_BASIC_INFO_NOT_ALUMNI";
-
+        public static string ALL_BASIC_INFO_NOT_ALUM = 
+            "select DISTINCT A.firstname as Firstname, A.lastname as Lastname, " +
+            "COALESCE(B.NickName, C.Nickname) as Nickname, " +
+            "COALESCE(B.MaidenName, C.MaidenName) as MaidenName, A.AD_Username as Username, " +
+            "CONCAT(A.firstname, A.lastname, A.AD_Username) as ConcatonatedInfo " +
+            "from ACCOUNT as A left join FacStaff as B on A.AD_Username = B.AD_Username " +
+            "left join Student as C on A.AD_Username = C.AD_Username " +
+            "where isnull(A.AD_Username, '') != '' " +
+            "and(isnull(B.AD_Username, '') != '' " +
+            "or isnull(C.AD_Username, '') !='')";
```

Comment:
Why are you using `left join` instead of `inner join` here? The check for non-null facStaff username or student username has the same effect as inner joining would, but in a more confusing (and possibly less performant) way.

Reply:
That's a good point! Thank you for suggesting that! I will try that tomorrow morning.

Reply:
After discussing the query with @jason-whitehouse, we determined that the best solution is probably the following:
```sql
from account
where AD_Username in (Select distinct AD_Username from student union Select distinct AD_Username from facstaff)
```

The problem with inner joining, we realized, is that it would require rows to match AD_Username in both FacStaff AND Student (which is impossible because those are mutually-exclusive subsets of Account). This achieves the desired behavior of selecting rows from Account as long as they match a row in either FacStaff or Student. It is also more concise and straightforward, and (depending on how sql executes it), it should be at least as performant if not moreso.

Reply:
I wonder if this solution checks whether the AD_Username is null. We don't want to find anyone whose AD_Username is null on the quick people search, because even we find them, they still don't have their profiles.

Reply:
You can simply add a non-null filter to the where clause, e.g. 
```sql
where AD_Username in (...) AND AD_Username IS NOT NULL
```

Reply:
There is also another concern. The front end requires NickName and MaidenName to be returned as part of the result. Both NickName and MaidenName only exist in Student and FacStaff. So we have to join Account with these two databases to get the info we need. The union is a really good idea, but it only restricts the AD_Username instead of getting the required data from other databses.